dev/unix/notion-install.sh

```diff
@@ -0,0 +1,378 @@
+  local install_dir="$2"
+
+  notion_info 'Checking' "for existing Notion installation"
+  if notion_upgrade_is_ok "$version" "$install_dir"
+  then
+    download_archive="$(notion_download_release "$version"; exit "$?")"
+    exit_status="$?"
+    if [ "$exit_status" != 0 ]
+    then
+      notion_error "Could not download Notion version '$version'\n\nSee https://github.com/notion-cli/notion/releases for a list of available releases"
+      return "$exit_status"
+    fi
+
+    notion_install_from_file "$download_archive" "$install_dir"
+  fi
+}
+
+notion_install_local() {
+  local dev_or_release="$1"
+  local install_dir="$2"
+
+  # compile and package the binaries, then install from that local archive
+  local _compiled_archive="$(notion_compile_and_package "$dev_or_release")"
+  notion_install_from_file "$_compiled_archive" "$install_dir"
+}
+
+notion_compile_and_package() {
+  local dev_or_release="$1"
+  # TODO: call the release script to do this, and return the packaged archive file
+  # TODO: parse the output to get the archive file name
+  dev/unix/release.sh "--$dev_or_release"
+  # TODO: check exit status
+  echo "target/release/notion-0.3.0-macos.tar.gz"
+}
+
+notion_download_release() {
+  local _version="$1"
+
+  local uname_str="$(uname -s)"
+  local openssl_version="$(openssl version)"
+  local os_info="$(parse_os_info "$uname_str" "$openssl_version")"
+  local pretty_os_name="$(parse_os_pretty "$uname_str")"
+
+  notion_info 'Fetching' "archive for $pretty_os_name, version $_version"
+
+  # store the downloaded archive in a temporary directory
+  local _download_dir="$(mktemp -d)"
+  local _filename="notion-$_version-$os_info.tar.gz"
+  local _download_file="$_download_dir/$_filename"
+
+  # TODO: for now, download the test files from my desktop
+  local notion_archive="http://mistewar-ld2.linkedin.biz:8080/$_filename"
+  # this will eventually be
+  # local notion_archive="https://github.com/notion-cli/notion/releases/download/v$_version/$_filename"
+
+  curl --progress-bar --show-error --location --fail "$notion_archive" --output "$_download_file" && echo "$_download_file"
+}
+
+notion_install_from_file() {
+  local _archive="$1"
+  local _extract_to="$2"
+
+  notion_info 'Creating' "directory layout"
+  notion_create_tree "$_extract_to"
+
+  notion_info 'Extracting' "Notion binaries and launchers"
+  # extract the files to the specified directory
+  echo "running: 'tar -xzvf "$_archive" -C "$_extract_to"'" >&2
+  tar -xzvf "$_archive" -C "$_extract_to"
+}
+
+# return if sourced (for testing the functions)
+return 0 2>/dev/null
+
+# TODO: do I actually want this?
+# exit on error
+# set -e
+
+# default to installing the latest available Notion version
+install_version="latest"
+
+# install to NOTION_HOME, defaulting to ~/.notion
+install_dir="${NOTION_HOME:-"$HOME/.notion"}"
+
+# parse command line options
+while [ $# -gt 0 ]
+do
+  arg="$1"
+
+  case "$arg" in
+    -h|--help)
+      usage
+      exit 0
+      ;;
+    --dev)
```

Comment:
In `usage()` above, this is named `--debug`.

Reply:
ah, good catch! I will change that